### ISSUE TITLE
Extending this module to 3D

### ISSUE THREAD

**Issue by Foreverinne** [ybubnov/torch_delaunay#48]
Thanks for your excellent work. I would like to ask if this method can be adapted to 3D Delauney triangulation and whether it is differentiable？

**Comment by ybubnov** [comment]
Hi @Foreverinne, thank you for your interest to the project and the questions!

1. Implementation of `shull2d` uses geometry predicates for a 2-dimensional space, they are not directly translatable to 3-dimensional spaces.

In your case, are you looking for a surface triangulation for 3-dimensional objects or for a tetrahedrization (covering the 3d space with tetrahedra)?

2. When you refer to the different ability of the triangulation, what do you mean precisely, what is you use case? My current comprehension is that result of `shull2d` is an index of input points that comprise triangles. Index per-se have not notion of differential, but gradient of points could be acquired through indexing operation like following:

```py

points = torch.randn(100, 2, requires_grad=True)
points = points * 2
simplices = shull2d(points)

points[simplices].grad
```
